src/StorageCache.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace BlackBox;
+
+use IteratorAggregate;
+
+/**
+ * Caches an storage with another storage.
+ *
+ * @author Carlos Lombarte <lombartec@gmail.com>
+ */
+class StorageCache implements IteratorAggregate, Storage
+{
+    /**
+     * The storage that will be cached.
+     *
+     * @var Storage
+     */
+    private $sourceStorage;
+
+    /**
+     * The cache used to storage the decorated storage.
+     *
+     * @var Storage
+     */
+    private $storageCache;
+
+    /**
+     * @param Storage $sourceStorage
+     * @param Storage $storageCache
+     */
+    public function __construct(Storage $sourceStorage, Storage $storageCache)
+    {
+        $this->sourceStorage    = $sourceStorage;
+        $this->storageCache     = $storageCache;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($id)
+    {
+        $cachedData = $this->storageCache->get($id);
+
+        if (null === $cachedData) {
+            return $this->sourceStorage->get($id);
```

Comment:
Ah please add a test for that that way we know there will be no regression.

E.g. set then get (should get from source storage) then get again (should get from cache).